core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -138,21 +138,25 @@ object ConsoleConsumer extends Logging {
             throw e
           }
       }
-      checkErr(formatter)
+      if (checkErr(output, formatter)) {
+        // Consumer will be closed
+        return
```

Comment:
Overall this seems fine, but this should probably `throw` instead as it makes fewer assumptions about the structure of the calling code.